cli/src/main/java/hudson/cli/CLI.java

```diff
@@ -189,8 +189,11 @@ private Channel connectViaCliPort(URL jenkins, CliPort clip) throws IOException
                 rsp.write(ch);
             }
             String head = new BufferedReader(new StringReader(rsp.toString("ISO-8859-1"))).readLine();
-            if (!head.startsWith("HTTP/1.0 200 "))
-                throw new IOException("Failed to establish a connection through HTTP proxy: "+rsp);
+            if (!(head.startsWith("HTTP/1.0 200 ") || head.startsWith("HTTP/1.1 200 "))) {
+                s.close();
+                System.err.println("Failed to tunnel the CLI port through the HTTP proxy. Falling back to HTTP.");
```

Comment:
Don't print directly to stderr, i'm using CLI programmatically and don't need to have stderr on jvm.